### PR TITLE
Fix `mapdamage_rescaling` naming of output bam files

### PR DESCRIPTION
**Pull request by IdoBar**
Use `${bam.basename}` (or `${base}`, which was already defined but not used) to strip the `.bam` extension of the original input file for the output of `mapdamage_rescaling`, so filenames won't turn up like this: 
```
library_libmerged_rg_rmdup.bam_rescaled.bam
```

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
 - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/eager/tree/master/.github/CONTRIBUTING.md)
 - [ ] If necessary, also make a PR on the nf-core/eager _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### `Fixed`
 
+- Fix filename for bam output of `mapdamage_rescaling`
```
Comment (IdoBar) [inline CHANGELOG.md:14]:
```suggestion
- [#719](https://github.com/nf-core/eager/pull/719) - Fix filename for bam output of `mapdamage_rescaling`
```

**Review by jfy133 — APPROVED** [review]

**Comment by jfy133** [comment]
Just in time for a new patch release! Thank you!!

**Comment by IdoBar** [comment]
No worries @jfy133, I have a few other ideas for improvements (which I should submit as issues), I wish I had the time/energy to stay awake for the NF hackathon and do those...
I'll just keep pushing those one by one when I get a chance.

**Comment by jfy133** [comment]
Awesome! Feel free to bring them up in the slack channel if you want informal chat/feedback about them already

**Pull request merged**